docs/contributor/07-module-migration-guide.md

```diff
@@ -150,21 +304,106 @@ The migration strategy involves replicating the current state with the NEW metad
 
 > [!Note]
 > After rollback, you can still use the old submission pipeline to submit new versions of the module while you're working on a fix.
 
-9. Delete all old metadata files related to the module.
+5. Promote to `stage` landscape
+
+   Once verified on `dev`, promote the new metadata to `stage`.
+
+   To do so, create a PR to `/kyma/kyma-modules` adding the `stage` landscape to the `targetLandscapes` in `/module-manifests/modules/<module-name>/module-releases.yaml`. For example:
+
+   ```yaml
+   targetLandscapes:
+     - dev
+     - stage # <= stage landscape added
+   channels:
+     - channel: regular
+       version: 1.34.0
+     - channel: fast
+       version: 1.34.0
+     - channel: experimental
+       version: 1.34.0-experimental
+     - channel: dev
+       version: 1.35.0-rc1
+   ```
+
+6. Verify if the module is updated in KCP.
+
+   ArgoCD picks up this change and deploys the new ModuleReleaseMeta to the `stage` landscape. KLM is now picking up the version change and updating all modules using the `regular` channel to version `1.34.0`.
+
+7. [OPTIONAL] Roll back the new metadata.
+
+   In case of failure, you can revert the setup to the old approach.
+
+   To do so, a PR can be opened to `/kyma/kyma-modules` reverting the submissions from steps 5. of Replicating the current state and 5. of Performing a version upgrade. It is important to revert completely removing the entire new metadata from KCP so that KLM falls back to the old module metadata.
+
+> [!Note]
+> After rollback, you can still use the old submission pipeline to submit new versions of the module while you're working on a fix.
+
+8. Promote to `prod` landscape
+
+   Once verified on `stage`, promote the new metadata to `prod`.
+
+   To do so, create a PR to `/kyma/kyma-modules` adding the `prod` landscape to the `targetLandscapes` in `/module-manifests/modules/<module-name>/module-releases.yaml`. For example:
+
+   ```yaml
+   targetLandscapes:
+     - dev
+     - stage
+     - prod # <= prod landscape added
+   channels:
+     - channel: regular
+       version: 1.34.0
+     - channel: fast
+       version: 1.34.0
+     - channel: experimental
+       version: 1.34.0-experimental
+     - channel: dev
+       version: 1.35.0-rc1
+   ```
+
+9. Verify if the module is updated in KCP.
+
+   ArgoCD picks up this change and deploys the new ModuleReleaseMeta to the `prod` landscape. KLM is now picking up the version change and updating all modules using the `regular` channel to version `1.34.0`.
+
+10. [OPTIONAL] Roll back the new metadata.
+
+   In case of failure, you can revert the setup to the old approach.
+
+   To do so, a PR can be opened to `/kyma/kyma-modules` reverting the submissions from steps 8. of Replicating the current state and 8. of Performing a version upgrade. It is important to revert completely removing the entire new metadata from KCP so that KLM falls back to the old module metadata.
```

Comment:
```suggestion
   To roll back, open a PR to `/kyma/kyma-modules` reverting the submissions to stage from **both** [Replicating the Current State](#replicating-the-current-state) and [Performing a Version Upgrade]. It is important to revert completely removing the entire new metadata from KCP so that KLM falls back to the old module metadata.
```
As in line 317, consider using a warning if this could lead to bad trouble.